base/sharedarray.jl

```diff
@@ -24,6 +24,17 @@ type SharedArray{T,N} <: DenseArray{T,N}
     SharedArray(d,p,r,sn) = new(d,p,r,sn)
 end
 
+call{T,N}(::Type{SharedArray{T}}, d::NTuple{N,Int}; kwargs...) =
+    SharedArray(T, d; kwargs...)
+call{T}(::Type{SharedArray{T}}, d::Integer...; kwargs...) =
+    SharedArray(T, d; kwargs...)
+call{T}(::Type{SharedArray{T}}, m::Integer; kwargs...) =
```

Comment:
My original intention was to make ShareArray have the same interface as Array, since I am writing a library using both with type parameters. My fix is kind of naive. And Kevin Squire said in the user group that there could be deeper and cleaner than falling back to SharedArray(T, x), maybe someone could improve later.